### PR TITLE
Chore: Added Questionnaire for dropdown type

### PR DESCRIPTION
**Pull request by preethijayanthi**

### CODE REVIEW

File: src/app/modules/consultations/consultation-profile/read-respond/read-respond.component.html [hunk 1]
```diff
@@ -70,16 +70,23 @@
...
+                      <input type="radio" class="cm-radio" (input)="onChange(); onAnswerChange(question,subQuestion)" [formControlName]="question.id" [value]="subQuestion.id">
                       <span>{{subQuestion.questionText}}</span>
                     </label>
                   </div>
                 </div>
+                <div class="answer" *ngIf="question.questionType === 'dropdown'">
+                    <ng-select 
+                    [items]="question.subQuestions" bindLabel="questionText" [virtualScroll]="true" dropdownPosition="bottom" 
+                    bindValue="id"  [clearable]="false" loadingText="Searching..." [searchable]="false" (change)="onChange();onAnswerChange(question, $event)"
+                    placeholder="Select" [formControlName]="question.id" required>
```
Comment (avikt) [inline src/app/modules/consultations/consultation-profile/read-respond/read-respond.component.html:82]:
```suggestion
                    placeholder="Select an option" [formControlName]="question.id" required> 
```

File: src/app/modules/consultations/consultation-profile/read-respond/read-respond.component.ts [hunk 2]
```diff
@@ -186,6 +209,25 @@ export class ReadRespondComponent implements OnInit, AfterViewChecked {
...
 
+  onAnswerChange(question?, value?) {
+    if (question && value.id === 'other') {
+      for (let i = 0; i < this.profileData.questions.length; i++) {
+        if (this.profileData.questions[i].id === question.id) {
+          this.profileData.questions[i].is_other = true;
+          break;
+        }
+      }
+    } else {
```
Comment (avikt) [inline src/app/modules/consultations/consultation-profile/read-respond/read-respond.component.ts:220]: @preethijayanthi What is this block for?

File: src/app/modules/consultations/consultation-profile/read-respond/read-respond.component.ts [hunk 3]
```diff
@@ -158,20 +159,42 @@ export class ReadRespondComponent implements OnInit, AfterViewChecked {
...
-            question_id: item,
-            answer: answers[item]
-          });
+          if (answers[item] === 'other') {
+            value.push({
+              question_id: item,
+              is_other: true,
+              other_option_answer: answers['other_answer-' + item],
+            });
+          } else if (!(item.includes('other'))) {
```
Comment (avikt) [inline src/app/modules/consultations/consultation-profile/read-respond/read-respond.component.ts:188]: @preethijayanthi I think this block is still running for answers with other. I tested with Consultation 155 on staging and the response object had a duplicate- 
image

Comment (avikt) [reply src/app/modules/consultations/consultation-profile/read-respond/read-respond.component.ts:188]: Also when only the other option is selected, we do not need to send `answer: []` like here- 
image

**Review by avikt — CHANGES_REQUESTED** [review]
Left a few feedback points

File: src/app/modules/consultations/consultation-profile/read-respond/read-respond.component.ts (hunk 2 above)
Comment (preethijayanthi) [reply src/app/modules/consultations/consultation-profile/read-respond/read-respond.component.ts:229]: When I click other option and click some other option to hide the input box.

Comment (avikt) [reply src/app/modules/consultations/consultation-profile/read-respond/read-respond.component.ts:229]: @preethijayanthi I think one more improvement can be made here. Right now in the case of checkbox type questions, even if the other option is unchecked, the input box still there. If others option is unchecked in a checkbox type question, the others input box should also be hidden

**Review by avikt — APPROVED** [review]

**Pull request merged**